sklearn/feature_selection/_rfe.py

```diff
@@ -811,3 +912,39 @@ def fit(self, X, y, groups=None):
             "n_features": step_n_features_rev,
         }
         return self
+
+    def get_metadata_routing(self):
+        """Get metadata routing of this object.
+
+        Please check :ref:`User Guide <metadata_routing>` on how the routing
+        mechanism works.
+
+        .. versionadded:: 1.6
+
+        Returns
+        -------
+        routing : MetadataRouter
+            A :class:`~sklearn.utils.metadata_routing.MetadataRouter` encapsulating
+            routing information.
+        """
+        router = MetadataRouter(owner=self.__class__.__name__)
+        router.add(
+            estimator=self.estimator,
+            method_mapping=MethodMapping().add(caller="fit", callee="fit"),
+        )
+        router.add(
+            splitter=check_cv(self.cv),
+            method_mapping=MethodMapping().add(
+                caller="fit",
+                callee="split",
+            ),
+        )
+        router.add(
+            scorer=self._get_scorer(),
+            method_mapping=MethodMapping().add(caller="fit", callee="score"),
```

Comment:
shouldn't `score` also call `score`?

Reply:
Does this have a score method?

Reply:
Yes, I would consider this as a bug and fix it here and add an entry in the changelog stating that the estimator is now using the scoring metadata when calling `score`.

We had this pattern in `LogisticRegressionCV` and I assume that we considered it as a bug as well.

Reply:
Do you mean we need to add a score method which uses the scorer?

Reply:
Yep similarly to:

https://github.com/scikit-learn/scikit-learn/blob/2ffe206719a7c45f550e5e68ffc69c189660754f/sklearn/linear_model/_logistic.py#L2190-L2235